### PR TITLE
Case Tile styling initial refactoring to help with isolating changes

### PR DESCRIPTION
**Pull request by biyeun**
## Technical Summary
This shouldn't have any visible effect on the current case tile usage. Review commit-by-commit. Tried my best to preserve history in the refactors.

- isolated style changes to case tiles so that it's easier to understand where the overrides and styles are coming from. also so that future styles are in a consistent place.
- refactored some of the html of the `case_list` template so that it's a little easier to follow around all the logic

### Safety story
Putting this on `staging` and using it alongside my test overrides of case tiles.

### Automated test coverage
No

### QA Plan
Curious how frequent QA requests are for the case tiles just to rule out any issues?

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by orangejenny — APPROVED** [review]
These look great. My one suggestion is to rename `case-tile.less`, because the styles aren't all specific to case tiles. Maybe `case-list.less`?

I'm not that concerned about QA for this, given that all of the changes are either moving code blocks between LESS files or extracting HTML into templates, but I'd request that you wait until after tomorrow morning's deploy to merge, just so it's not slipping into the deploy last minute.

File: corehq/apps/hqwebapp/static/cloudcare/less/formplayer-webapp/case-tile.less
Comment (MartinRiese) [inline corehq/apps/hqwebapp/static/cloudcare/less/formplayer-webapp/case-tile.less:123]: I think this would be a little bit easier to understand:
```
#menu-region {
  .module-menu-container, .module-case-list-container {
  }
}
```

**Comment by MartinRiese** [comment]
QA will also do a pass over a bunch of [BHA related changes on the 12th](https://dimagi.atlassian.net/browse/QA-6171). So if this is up on staging at that time it will get a passive QA treatment.

**Review by nospame — APPROVED** [review]

**Comment by biyeun** [comment]
@MartinRiese were any changes flagged during QA that would be concerning for this PR? I was heads down for a design sprint last week. I think it's safe to merge cc @orangejenny

**Comment by MartinRiese** [comment]
No, QA did not find anything concerning.

**Pull request merged**
